mbuild/compound.py

```diff
@@ -1891,16 +1897,21 @@ def to_trajectory(self, show_ports=False, chains=None,
             xyz[0, idx] = atom.pos
 
         # Unitcell information.
-        box = self.boundingbox
-        unitcell_lengths = np.empty(3)
-        for dim, val in enumerate(self.periodicity):
-            if val:
-                unitcell_lengths[dim] = val
-            else:
-                unitcell_lengths[dim] = box.lengths[dim]
+        unitcell_angles = [90.0, 90.0, 90.0]
+        if box is None:
+            box = self.boundingbox
+            unitcell_lengths = np.empty(3)
+            for dim, val in enumerate(self.periodicity):
+                if val:
+                    unitcell_lengths[dim] = val + 0.5
```

Comment:
I think you have this flipped. This should be `unitcell_lengths[dim] = val` (so that's its the same length as the value from the `periodicity` attribute) and the `else` case should be `unitcell_lengths[dim] = box.lengths[dim] + 0.5`. Looks like we also need some more thorough unit tests as well...